### DISCUSSION TITLE
how do I maintain the input image sizes I've manually sized?

### DISCUSSION THREAD

**Discussion by xings19** [bghira/SimpleTuner#560] [Q&A] [answered]
The image I input is 512 in width and 2048 in height. I also want the image to keep this size for training. How should I set the resolution-related parameters? I have read the document carefully, but I still don't understand some of the settings. 
This is my `multidatabackend.json`
```json
[
    {
      "id": "my_dataset",
      "type": "local",
      "crop": false,
      "cache_dir_vae": "cache/vae/sd3/my_data",
      "instance_data_dir": "a/local/dir",
      "disabled": false,
      "skip_file_discovery": "",
      "caption_strategy": "textfile",
      "metadata_backend": "json"
    },
    {
      "id": "text-embeds",
      "type": "local",
      "dataset_type": "text_embeds",
      "default": true,
      "cache_dir": "cache/text/sd3/my_data",
      "disabled": false,
      "write_batch_size": 1
    }
  ]
```

**Comment by xings19** [comment]
if i change the json file to:
```json
[
    {
      "id": "my_dataset",
      "type": "local",
      "crop": false,
      "resolution": 512,
      "resolution_type": "pixel",
      "cache_dir_vae": "cache/vae/sd3/my_data",
      "instance_data_dir": "a/local/dir",
      "disabled": false,
      "skip_file_discovery": "",
      "caption_strategy": "textfile",
      "metadata_backend": "json"
    },
    {
      "id": "text-embeds",
      "type": "local",
      "dataset_type": "text_embeds",
      "default": true,
      "cache_dir": "cache/text/sd3/my_data",
      "disabled": false,
      "write_batch_size": 1
    }
  ]
```
Is this correct?

**Comment by bghira** [comment]
yes - use

```bash
export SIMPLETUNER_DEBUG_IMAGE_PREP=true
```

on the command line just before running the trainer.

it will write all processed images at each step of their being processed to `SimpleTuner/images` (you may have to create the dir first, or it will error out.)

if you observe any issue, please ping me

**Comment by bghira** [comment, marked as answer]
also, remember to disable / `unset` that env var for your actual training, or it might keep writing images and fill your disk up :-) it's useful for verifying that the images are undisturbed though.

**Reply by xings19** [reply]
Thanks!

**Reply by xings19** [reply]
Actually, if I set up `multidatabackend.json` as mentioned before, setting this environment variable has no use

**Reply by bghira** [reply]
no, it should still write out the images during each step of their preparation

**Reply by xings19** [reply]
Ok, i know, i must delete the vae cache of image first, then the variable will work, thanks!

**Reply by bghira** [reply]
ohh - yes. the images are only prepared before they are encoded.
